### PR TITLE
MP indexer works 

### PR DESCRIPTION
**Pull request by IlincaTudose**

### CODE REVIEW

File: data-model/pom.xml [hunk 1]
```diff
@@ -90,6 +90,11 @@
             <artifactId>owlapi-distribution</artifactId>
             <version>4.1.4</version>
         </dependency>
+        <dependency>
```
Comment (IlincaTudose) [inline data-model/pom.xml]: still using this dependency?

File: data-model/src/main/java/org/mousephenotype/cda/owl/OntologyParser.java [hunk 2]
```diff
@@ -50,20 +50,72 @@ public OntologyParser(String pathToOwlFile, String prefix, Set<String> topLevelI
...
+
+    /**
+     * Computes paths in the format needed for TreeJs, for the ontology browsers.
+     * [!] This is not computed by default. If you want the trees, call this method on the parser first.
+     */
+    public void fillJsonTreePath(String rootId){
+
+        OWLClass root = classMap.get(rootId);
+        fillJsonTreePath(root, new ArrayList<>());
+        // TODO Produce json obj
```
Comment (IlincaTudose) [inline data-model/src/main/java/org/mousephenotype/cda/owl/OntologyParser.java]: remove todo comments (done elsewhere)

Comment (IlincaTudose) [inline data-model/src/main/java/org/mousephenotype/cda/owl/OntologyParser.java]: Do JSONs here, not in OntologyBrowserGetter

File: indexers/src/main/java/org/mousephenotype/cda/indexers/MPIndexer.java [hunk 3]
```diff
@@ -176,15 +143,22 @@ public RunStatus run ()
...
         initialiseSupportingBeans();
-        Set<String> wantedIds = getWantedSlimIds();
+        OntologyBrowserGetter browser = new OntologyBrowserGetter();
 
         try {
-            OntologyParser mpParser = new OntologyParser(owlpath + "/mp.owl", "MP", TOP_LEVEL_MP_TERMS, wantedIds);
+            mpParser = getMpParser();
+            mpParser.fillJsonTreePath("MP:0000001"); // call this if you want node ids from the objects
+            System.out.println("Loaded mp parser");
             mpHpParser = new OntologyParser(owlpath + "/mp-hp.owl", "MP", null, null);
```
Comment (jmason-ebi) [inline indexers/src/main/java/org/mousephenotype/cda/indexers/MPIndexer.java]: for consistency move this to the abstract indexer with a getter like the others

File: indexers/src/main/java/org/mousephenotype/cda/indexers/PipelineIndexer.java [hunk 4]
```diff
@@ -94,8 +95,9 @@ public RunStatus validateBuild()	throws IndexerException {
 
 
 	private void initialiseSupportingBeans(RunStatus runStatus)
-			throws IndexerException, SQLException {
+			throws IndexerException, SQLException, OWLOntologyCreationException, OWLOntologyStorageException, IOException {
```
Comment (jmason-ebi) [inline indexers/src/main/java/org/mousephenotype/cda/indexers/PipelineIndexer.java]: Catch SQLException, OWLOntologyCreationException, OWLOntologyStorageException, IOException and wrap into IndexerException

File: indexers/src/main/java/org/mousephenotype/cda/indexers/PipelineIndexer.java [hunk 5]
```diff
@@ -68,10 +68,11 @@
 	private Map<String, ParameterDTO> paramIdToParameter;
 	private Map<String, ProcedureDTO> procedureIdToProcedure;
 	private Map<String, PipelineDTO> pipelines;
-	private Map<String, MpDTO> mpIdToMp;
 	private Map<String, ObservationType> parameterToObservationTypeMap;
 	private Map<String, EmapaOntologyDAO.Emapa> emap2EmapaMap;
 
+	private OntologyParser mpParser;
+
 	private static Connection ontoDbConnection;
```
Comment (jmason-ebi) [inline indexers/src/main/java/org/mousephenotype/cda/indexers/PipelineIndexer.java]: Can remove :)

File: indexers/src/main/java/org/mousephenotype/cda/indexers/PipelineIndexer.java [hunk 6]
```diff
@@ -106,8 +108,6 @@ private void initialiseSupportingBeans(RunStatus runStatus)
         ontoDbConnection = ontodbDataSource.getConnection();
```
Comment (jmason-ebi) [inline indexers/src/main/java/org/mousephenotype/cda/indexers/PipelineIndexer.java]: Can remove ontodbConnection now?

**Comment by jmason-ebi** [comment]
Reviewed.  Ready to merge when issues are complete.

File: data-model/src/main/java/org/mousephenotype/cda/owl/OntologyParser.java (hunk 2 above)
Comment (IlincaTudose) [inline data-model/src/main/java/org/mousephenotype/cda/owl/OntologyParser.java]: done

File: indexers/src/main/java/org/mousephenotype/cda/indexers/PipelineIndexer.java (hunk 5 above)
Comment (IlincaTudose) [inline indexers/src/main/java/org/mousephenotype/cda/indexers/PipelineIndexer.java]: removed now.

File: indexers/src/main/java/org/mousephenotype/cda/indexers/PipelineIndexer.java (hunk 4 above)
Comment (IlincaTudose) [inline indexers/src/main/java/org/mousephenotype/cda/indexers/PipelineIndexer.java]: done

**Pull request merged**